llmfoundry/models/layers/dmoe.py

```diff
@@ -279,7 +279,8 @@ def forward(
             topk_list = topk_idx.tolist()
 
             expert_tokens = x[None, token_list].reshape(-1, hidden_size)
-            mlp_output = self.mlp(expert_tokens, expert_idx)
+            mlp_output = self.mlp(expert_tokens,
+                                  expert_idx).to(expert_weights.device)
```

Comment:
same comment
> the fact that this one is needed is scary...

Reply:
Discussed offline. Modified the code accordingly.